src/configgen.py

```diff
@@ -717,26 +733,32 @@ def parseFooterDoc(node):
             if (n.nodeName == "docs"):
                 if (n.getAttribute('documentation') != "0"):
                     doc += parseDocs(n)
-    print(doc)
+    xprint(doc)
 
 
 def main():
     if len(sys.argv)<3 or (not sys.argv[1] in ['-doc','-cpp','-wiz','-maph','-maps']):
         sys.exit('Usage: %s -doc|-cpp|-wiz|-maph|-maps config.xml' % sys.argv[0])
     try:
-        doc = xml.dom.minidom.parse(sys.argv[2])
+        if sys.version_info[0] == 2:
+            fh = open(sys.argv[2],'r')
+            sys.stdout = io.open(sys.stdout.fileno(), 'w')
```

Comment:
I think also python 2.7 supports encoding when using `io.open`, i.e. this seems to work fine:

```python
import io
import sys
fh = io.open('file.txt','r',encoding='utf-8')
sys.stdout = io.open(sys.stdout.fileno(), 'w', encoding='utf8')
sys.stdout.write(fh.read())
```

Reply:
This looks like to work for Python 2 and Python 3, do you want me to patch this or will you do it?

Reply:
The first test was with the default config.xml, I now ttried with the Chinese version and this gives problems:
```
[ 15%] Generating ../generated_src/configoptions.cpp

'ascii' codec can't encode characters in position 8373-8391: ordinal not in range(128)
```


So I think my original patch is OK.
```